envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/SanUriMatcherFactory.kt

```diff
@@ -8,17 +8,37 @@ import java.lang.IllegalArgumentException
 class SanUriMatcherFactory(
     private val tlsProperties: TlsAuthenticationProperties
 ) {
+    private val luaEscapeRegex = Regex("""[-().%+*?\[^$]""")
```

Comment:
Why do we need to escape anything here?

Reply:
The same reason we have to escape it for RegexMatcher in Envoy (this is also in this file - SanUriMatcherFactory): because the format is a literal string so we don't want to interpret it as a regex.

Look at the test (SanUriMatcherFactoryTest). We have configuration:

```            
isanUriFormat = "spiffe://{service-name}?env=dev"
serviceNameWildcardRegex = ".+"
```

and the expected result is: 
```
^spiffe://(.+)%?env=dev$
```
You can see, we have to add `%` before `?`, otherwise `?` will be treated as a regex special character. And this configuration should look familiar to you :)